packages/celotool/src/e2e-tests/governance_tests.ts

```diff
@@ -128,31 +136,115 @@ describe('governance tests', () => {
+    signerWeb3: any,
+    txOptions: any = {}
+  ) => {
+    const validator = (await validatorWeb3.eth.getAccounts())[0]
+    const signer = (await signerWeb3.eth.getAccounts())[0]
+    await unlockAccount(validator, validatorWeb3)
+    await unlockAccount(signer, signerWeb3)
+    const pop = await (await newKitFromWeb3(
+      signerWeb3
+    ).contracts.getAccounts()).generateProofOfSigningKeyPossession(validator, signer)
+    const validatorKit = newKitFromWeb3(validatorWeb3)
+    const validatorAccounts = await validatorKit._web3Contracts.getAccounts()
+    const tx = validatorAccounts.methods.authorizeValidatorSigner(signer, pop.v, pop.r, pop.s)
+    let gas = txOptions.gas
+    if (!gas) {
+      gas = await tx.estimateGas({ ...txOptions })
+    }
+    return tx.send({ from: validator, ...txOptions, gas })
+  }
+
+  const updateValidatorBlsKey = async (
+    validatorWeb3: any,
+    signerWeb3: any,
+    signerPrivateKey: string,
+    txOptions: any = {}
+  ) => {
+    const validator = (await validatorWeb3.eth.getAccounts())[0]
+    const signer = (await signerWeb3.eth.getAccounts())[0]
+    await unlockAccount(signer, signerWeb3)
+    const blsPublicKey = getBlsPublicKey(signerPrivateKey)
+    const blsPop = getBlsPoP(validator, signerPrivateKey)
+    // TODO(asa): Send this from the signer instead.
+    const validatorKit = newKitFromWeb3(validatorWeb3)
+    const validatorValidators = await validatorKit._web3Contracts.getValidators()
+    const tx = validatorValidators.methods.updateBlsKey(blsPublicKey, blsPop)
+    let gas = txOptions.gas
+    if (!gas) {
+      gas = await tx.estimateGas({ ...txOptions })
+    }
+    return tx.send({ from: validator, ...txOptions, gas })
+  }
+
   const isLastBlockOfEpoch = (blockNumber: number, epochSize: number) => {
     return blockNumber % epochSize === 0
   }
 
   describe('when the validator set is changing', () => {
     let epoch: number
     const blockNumbers: number[] = []
-    let allValidators: string[]
+    let validatorAccounts: string[]
     before(async function(this: any) {
       this.timeout(0) // Disable test timeout
       await restart()
-      const [groupAddress, groupPrivateKey] = await getValidatorGroupKeys()
-
-      const groupInstance = {
-        name: 'validatorGroup',
-        validating: false,
-        syncmode: 'full',
-        port: 30325,
-        wsport: 8567,
-        privateKey: groupPrivateKey.slice(2),
-        peers: [await getEnode(8545)],
-      }
-      await initAndStartGeth(context.hooks.gethBinaryPath, groupInstance)
-      allValidators = await getValidatorGroupMembers()
-      assert.equal(allValidators.length, 5)
+      const groupPrivateKey = await getValidatorGroupPrivateKey()
+      const rotation0PrivateKey =
+        '0xa42ac9c99f6ab2c96ee6cae1b40d36187f65cd878737f6623cd363fb94ba7087'
+      const rotation1PrivateKey =
+        '0x4519cae145fb9499358be484ca60c80d8f5b7f9c13ff82c88ec9e13283e9de1a'
+      const additionalNodes: any[] = [
```

Comment:
`GethInstanceConfig` from `src/e2e-tests/util` would be more specific
```suggestion
      const additionalNodes: GethInstanceConfig[] = [
```